src/encode.rs

```diff
@@ -352,6 +312,42 @@ impl<'a, I: AsRef<[u8]>> EncodeBuilder<'a, I> {
     }
 }
 
+/// Encode into a new owned string.
+///
+/// # Examples
+///
+/// ```rust
+/// let input = [0x04, 0x30, 0x5e, 0x2b, 0x24, 0x73, 0xf0, 0x58];
+/// let output: String = bs58::encode(input).into();
+/// assert_eq!("he11owor1d", output);
+/// ```
+#[cfg(feature = "alloc")]
+impl<I: AsRef<[u8]>> From<EncodeBuilder<'_, I>> for String {
+    fn from(builder: EncodeBuilder<'_, I>) -> Self {
+        let mut output = String::new();
+        builder.onto(&mut output).unwrap();
+        output
+    }
+}
+
+/// Encode into a new owned vector.
+///
+/// # Examples
+///
+/// ```rust
+/// let input = [0x04, 0x30, 0x5e, 0x2b, 0x24, 0x73, 0xf0, 0x58];
+/// let output: Vec<u8> = bs58::encode(input).into();
+/// assert_eq!(b"he11owor1d".as_slice(), output);
+/// ```
+#[cfg(feature = "alloc")]
+impl<I: AsRef<[u8]>> From<EncodeBuilder<'_, I>> for Vec<u8> {
+    fn from(builder: EncodeBuilder<'_, I>) -> Self {
+        let mut output = Vec::new();
+        builder.onto(&mut output).unwrap();
+        output
+    }
+}
+
```

Comment:
Seems to me that we need a generic TryFrom implementation:

```rust
impl<'a, T, E> std::convert::TryFrom<EncodeBuilder<'a, T>> for E
where
    T: AsRef<[u8]>,
    E: EncodeTarget + Default
{
    type Error = Error;

    fn try_from(builder: EncodeBuilder<'a, I>) -> Result<Self> {
        let mut output = Default::default();
        builder.onto(&mut output)?;
        Ok(output)
    }
}
```

and then, for encoders which can allocate space, have another trait
and generic From implementation for that:

```rust
/// Marker trait to denote that callback passed to `encode_with` is guaranteed
/// to be given at least `max_len`-byte-long buffer to write data to.  This
/// means that `encode_with` function will never fail with
/// [`Error::BufferTooSmall`] error.
pub trait ResizableEncodeTarget : EncodeTarget { }

impl<'a, T, E> std::convert::From<EncodeBuilder<'a, T>> for E
where
    T: AsRef<[u8]>,
    E: ResizableEncodeTarget + Default
{
    fn from(builder: EncodeBuilder<'a, I>) -> Self {
        let mut output = Default::default();
        builder.onto(&mut output).unwrap();
    }
}
```